File: HouseM8-Flutter/housem8_flutter/lib/models/address.dart
class Address {
  String street;
  int streetNumber;
  String postalCode;
  String district;
  String country;

  Address(
      {this.street,
      this.streetNumber,
      this.postalCode,
      this.district,
      this.country});

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
        street: json["street"],
        streetNumber: json["streetNumber"],
        postalCode: json["postalCode"],
        district: json["district"],
        country: json["country"]);
  }

  @override
  String toString() {
    return '$street, Nº$streetNumber $postalCode $district, $country';
  }

  Map<String, dynamic> toJson() {
    return {
      'street': street,
      'streetNumber': streetNumber,
      'postalCode': postalCode,
      'district': district,
      'country': country,
    };
  }
}

File: HouseM8API/HouseM8API/Entities/Address.cs

namespace HouseM8API.Entities
{
    public class Address
    {
        public string Street { get; set; }
        public int StreetNumber { get; set; }
        public string PostalCode { get; set; }
        public string District { get; set; }
        public string Country { get; set; }

        public override string ToString()
        {
            return "" + Street + ", Nº" + StreetNumber + " " + PostalCode + " " + District + ", " + Country;
        }
    }
}
